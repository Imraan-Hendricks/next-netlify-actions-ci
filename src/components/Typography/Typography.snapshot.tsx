import { render } from '@testing-library/react';
import { Typography } from './Typography';

describe('Typography snapshot tests', () => {
  it('renders with expected children', () => {
    const { container } = render(<Typography>Text to display</Typography>);
    expect(container).toMatchSnapshot();
  });

  it('renders with specified element', () => {
    const { container } = render(
      <Typography as='h1'>Text to display</Typography>,
    );
    expect(container).toMatchSnapshot();
  });

  it('renders with specified variant', () => {
    const { container } = render(
      <Typography variant='h1'>Text to display</Typography>,
    );
    expect(container).toMatchSnapshot();
  });

  it('renders with the "bold" class when "bold" prop is true', () => {
    const { container } = render(<Typography bold>Text to display</Typography>);
    expect(container).toMatchSnapshot();
  });

  it('renders with the "thin" class when "thin" prop is true', () => {
    const { container } = render(<Typography thin>Text to display</Typography>);
    expect(container).toMatchSnapshot();
  });

  it('renders with specified custom className', () => {
    const { container } = render(
      <Typography className='custom-class'>Text to display</Typography>,
    );
    expect(container).toMatchSnapshot();
  });
});
