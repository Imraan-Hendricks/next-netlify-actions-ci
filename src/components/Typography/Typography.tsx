import { ClassValue, cn } from '@/utils/common-utils';
import { createElement } from 'react';

interface TypographyProps {
  as?: 'h1' | 'h2' | 'h3' | 'h4' | 'h5' | 'h6' | 'p' | 'span';
  bold?: boolean;
  children: string;
  className?: ClassValue;
  thin?: boolean;
  variant?: 'title' | 'subtitle' | 'h1' | 'largeprint' | 'body' | 'fineprint';
}

export function Typography({
  as = 'p',
  bold,
  children,
  className,
  thin,
  variant = 'body',
}: TypographyProps) {
  const styles = {
    root: 'font-inter',
    title: cn(
      'text-6xl sm:text-7xl',
      thin && 'font-light',
      bold && 'font-medium',
    ),
    subtitle: cn(
      'text-2xl sm:text-3xl font-medium',
      thin && 'font-normal',
      bold && 'font-semibold',
    ),
    h1: cn(
      'text-2xl sm:text-3xl font-medium',
      thin && 'font-normal',
      bold && 'font-semibold',
    ),
    largeprint: cn(
      'text-lg sm:text-xl leading-relaxed',
      thin && 'font-light',
      bold && 'font-medium',
    ),
    body: cn(
      'text-sm sm:text-base',
      thin && 'font-light',
      bold && 'font-medium',
    ),
    fineprint: cn(
      'text-xs sm:text-sm',
      thin && 'font-light',
      bold && 'font-medium',
    ),
  };

  return createElement(
    as,
    { className: cn(styles.root, styles[variant], className) },
    children,
  );
}
