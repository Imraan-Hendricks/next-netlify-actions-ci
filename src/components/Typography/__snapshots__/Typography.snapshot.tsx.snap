// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Typography snapshot tests renders with expected children 1`] = `
<div>
  <p
    class="font-inter text-sm sm:text-base"
  >
    Text to display
  </p>
</div>
`;

exports[`Typography snapshot tests renders with specified custom className 1`] = `
<div>
  <p
    class="font-inter text-sm sm:text-base custom-class"
  >
    Text to display
  </p>
</div>
`;

exports[`Typography snapshot tests renders with specified element 1`] = `
<div>
  <h1
    class="font-inter text-sm sm:text-base"
  >
    Text to display
  </h1>
</div>
`;

exports[`Typography snapshot tests renders with specified variant 1`] = `
<div>
  <p
    class="font-inter text-2xl sm:text-3xl font-medium"
  >
    Text to display
  </p>
</div>
`;

exports[`Typography snapshot tests renders with the "bold" class when "bold" prop is true 1`] = `
<div>
  <p
    class="font-inter text-sm sm:text-base font-medium"
  >
    Text to display
  </p>
</div>
`;

exports[`Typography snapshot tests renders with the "thin" class when "thin" prop is true 1`] = `
<div>
  <p
    class="font-inter text-sm sm:text-base font-light"
  >
    Text to display
  </p>
</div>
`;
